### PR TITLE
Split the ExchangeService interface into two and some more changes

### PR DESCRIPTION
**Pull request by skubarenko**

### CODE REVIEW

File: src/utils/identifier.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+export type Identified<T> = T & { readonly id: string; };
+
+export const isIdentified = <T>(instance: T): instance is Identified<T> => {
+    return !!instance && (instance as any).id && typeof (instance as any).id === 'string';
+};
```
Comment (maxima-net) [inline src/utils/identifier.ts]: Is this a guard? Shell we use the "Guard" as a name postifx here?

File: src/utils/identifier.ts [hunk 2]
```diff
@@ -0,0 +1,10 @@
+export type Identified<T> = T & { readonly id: string; };
+
```
Comment (maxima-net) [inline src/utils/identifier.ts:2]: Are you sure that id must have a string type, not a number?

**Review by maxima-net — CHANGES_REQUESTED** [review]

File: src/utils/identifier.ts (hunk 1 above)
Comment (skubarenko) [reply src/utils/identifier.ts]: Yes, it's a guard, but I think that the "Guard" postfix is redundant. See the `nodejs` guards:
```ts
export function isNullOrUndefined(object: any): object is null | undefined;
export function isNumber(object: any): object is number;
export function isObject(object: any): boolean;
```
and examples in [the official documentation](http://www.typescriptlang.org/docs/handbook/advanced-types.html) of typescript:
```ts
function isFish(pet: Fish | Bird): pet is Fish {
    return (<Fish>pet).swim !== undefined;
}
```

File: src/utils/identifier.ts (hunk 2 above)
Comment (skubarenko) [reply src/utils/identifier.ts:2]: No, I'm not 😄  Need to check API methods of popular services to select the type correctly

File: src/utils/identifier.ts (hunk 1 above)
Comment (maxima-net) [reply src/utils/identifier.ts]: Ok. I agree.

File: src/utils/identifier.ts (hunk 2 above)
Comment (maxima-net) [reply src/utils/identifier.ts:2]: Ok, Let's make it string. It's safer then number typer.

**Review by maxima-net — APPROVED** [review]

**Pull request merged**